core-primitives/extrinsics-factory/src/lib.rs

```diff
@@ -88,6 +88,15 @@ where
 	) -> Self {
 		ExtrinsicsFactory { genesis_hash, signer, nonce_cache, node_metadata_repository }
 	}
+
+	pub fn with_signer(&self, signer: Signer, nonce_cache: Arc<NonceCache>) -> Self {
```

Comment:
this allows the enclave to send extrinsics using arbitrary signers (closes #1466)